src/index.tsx

```diff
@@ -0,0 +1,115 @@
+import { StyleSheet, Text, View, I18nManager } from 'react-native'
+import { RotateAlphabets, getAlphabet } from './alphabets/alphabets';
+import { TextStyle, TextProps  } from 'react-native'
+
+type MeasureMap = { width: number, height: number }
+interface OptionalTickerProps {
+  duration?: number
+  textStyle?: TextStyle
+  textProps?: TextProps
+}
+interface TickProps extends OptionalTickerProps {
+  character: string
+  allCharacters: string[]
+  measurement: { width: number, height: number }
+}
+interface TickerProps extends OptionalTickerProps {
+  charSet?: string
+  value?: string
+  rotateItems?: string[]
+  children: Array<string>
+}
+
+const styles = StyleSheet.create({
+  row: {
+    flexDirection: I18nManager.isRTL ? 'row-reverse' : 'row',
+    overflow: 'hidden',
+  },
+  hide: { position: 'absolute', top: 0, left: 0, opacity: 0 },
+})
+
+const Tick = (props: TickProps) => {
+  const { character, duration, allCharacters } = props
+
+  const height = useSharedValue(props.measurement.height)
+  const width = useSharedValue(props.measurement.width)
+
+  const charactersStyle = useAnimatedStyle(() => {
+    'worklet'
+    const toTranslateY = allCharacters.indexOf(character) * height.value * -1
+    const translateY =  withTiming(toTranslateY, { duration })
+    return {
+      height: height.value,
+      transform: [{ translateY }]
+    }
+  })
+
+  const viewPortStyle = useAnimatedStyle(() => {
+    'worklet'
+    return {
+      width: withTiming(width.value, { duration: 1000 }),
+      height: height.value,
+      overflow: 'hidden'
+    }
+  })
+
+  return (
+    <Animated.View style={[viewPortStyle]} >
+      <Animated.View style={[charactersStyle]} >
+        {allCharacters.map((char, i) => (
+          <Animated.Text key={i} {...props.textProps} style={[props.textStyle, {height: height.value}]}>
+            {char}
+          </Animated.Text>
+        ))}
+      </Animated.View>
+    </Animated.View>
+  )
+}
+
+export const Ticker = (props: TickerProps) => {
+  const { children, duration = 500, textStyle = {}, textProps = {}, rotateItems, value = children[0] } = props
+  const [measured, setMeasured] = useState<boolean>(false)
+  const measureMap = useRef<{ [key: string]: MeasureMap }>({}).current
+
+  const allCharacters: string[] = rotateItems != null
+    ? [ ...new Set(rotateItems.join('').split('')) ]
+    : getAlphabet(RotateAlphabets, props.charSet ?? 'Keypad.Numpad')
+
+  const onLayout = (char: string) => (event: any) => {
+    if (measureMap == null) return
+    measureMap[char] = { ...event?.nativeEvent?.layout }
+    if (Object.keys(measureMap).length === allCharacters.length) {
+      setMeasured(true)
+    }
+  }
+
+  const hiddenChildren = useMemo(() => allCharacters.map((char, i) =>
+    <Text
+      {...props}
+      style={[props.textStyle, styles.hide]}
+      key={i}
+      onLayout={onLayout(char)}
+    >{char}</Text>
+  ), [ measureMap ])
```

Comment:
Ah. Move the `if (measureMap == null) return` check into the memoization function.

```ts
  const onLayout = (char: string) => (event: any) => {
    measureMap[char] = { ...event?.nativeEvent?.layout }
    if (Object.keys(measureMap).length === allCharacters.length) {
      setMeasured(true)
    }
  }

  const hiddenChildren = useMemo(() => allCharacters.map((char, i) => {
    if (measureMap == null) return null
    return <Text
      {...props}
      style={[props.textStyle, styles.hide]}
      key={i}
      onLayout={onLayout(char)}
    >{char}</Text>
  }), [ measureMap ])
```

This way you only create the children once the `measureMap` ref is ready.

Reply:
I like that idea, but you got the code a bit wrong, this is what I changed it into:

```
 const onLayout = (char: string) => (event: any) => {
    measureMap[char] = { ...event?.nativeEvent?.layout }
    if (Object.keys(measureMap).length === allCharacters.length) {
      setMeasured(true)
    }
  }

  const hiddenChildren = useMemo(() => {
    if (measureMap == null) return
    return allCharacters.map((char, i) =>
      <Text
        {...props}
        style={[props.textStyle, styles.hide]}
        key={i}
        onLayout={onLayout(char)}
      >{char}</Text>)
  }, [ measureMap ])
```